### PR TITLE
revises integration testing script to check for failure conditions

### PR DESCRIPTION
**Pull request by brennanMKE**
*Issue #, if available:*

*Description of changes:*

Adds checks for failure conditions in script used for integration testing. The updated script has been tested locally. It does not change the regular behavior of the script.

*Check points:*

- [ ] ~~Added new tests to cover change, if needed~~
- [x] All unit tests pass
- [x] All integration tests pass
- [ ] ~~Updated CHANGELOG.md~~
- [ ] ~~Documentation update for the change if required~~
- [x] PR title conforms to conventional commit style

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Scripts/generate-test-config.sh [hunk 1]
```diff
@@ -198,6 +198,7 @@ cmd_quiet_flag="--quiet"
 [[ -n $region ]] || die "'region' not specified"
 export AWS_DEFAULT_REGION=${region}
 
+[[ ! -z $(which jq) ]] || die "jq is missing: brew install jq"
```
Comment (palpatim) [inline Scripts/generate-test-config.sh:201]: The [`validate_dependencies` function](https://github.com/aws-amplify/aws-sdk-ios/pull/3517/files#diff-02697d3f7e59bcd8b9d7b6c10f286fbd108a92cb81d544a200f743df66577d3cL205) should already validate this. It's only necessary for the assume-role use case (which you should be using to generate the test config locally), so it isn't appropriate to validate every time. `validate_dependencies` is invoked as the [first step in `main`](https://github.com/aws-amplify/aws-sdk-ios/pull/3517/files#diff-02697d3f7e59bcd8b9d7b6c10f286fbd108a92cb81d544a200f743df66577d3cL298). If we're seeing failures, let's fix that invocation or function logic.

Comment (brennanMKE) [reply Scripts/generate-test-config.sh:201]: Yes, it failed for me so I will look at that function to update it.

Comment (brennanMKE) [reply Scripts/generate-test-config.sh]: I've determined this check will not be needed and removed it in the next commit.

File: Scripts/generate-test-config.sh [hunk 2]
```diff
@@ -253,14 +252,26 @@ function resolve_credentials {
...
-    circleci_execution_role_arn=$(aws ssm get-parameter --name "/mobile-sdk/${platform}/common/circleci_execution_role" | jq -r .Parameter.Value)
+    circleci_execution_role_arn=$(echo "${circleci_execution_role}" | jq -r .Parameter.Value)
     readonly circleci_execution_role_arn
 
     local assume_role_creds
     assume_role_creds=$(aws sts assume-role --duration-seconds 3600 --role-arn "${circleci_execution_role_arn}" --role-session-name "IntegTest-$(date +%Y%m%d%H%M%S)")
     readonly assume_role_creds
 
+    if [ -z "${assume_role_creds}" ]; then
+      die "Failed to assume role credentials for Circle CI"
```
Comment (palpatim) [inline Scripts/generate-test-config.sh:272]: nit: error clarification
```suggestion
      die "Failed to assume Circle CI execution role"
```

**Review by palpatim — APPROVED** [review]
Minor error message update, otherwise LGTM

**Pull request merged**